lib/libyul2llvm/YulASTVisitor/CodegenVisitor.cpp

```diff
@@ -153,12 +153,13 @@ void LLVMCodegenVisitor::visitYulContractNode(YulContractNode &node) {
   allocateMemoryFunction = funDefHelper->createMemoryAllocatorFunction();
   allocateStorageFunction = funDefHelper->createStorageAllocatorFunction();
   runFunctionDeclaratorVisitor(node);
-  constructSelfStructType(node);
+  constructStructs(node);
   getModule().setSourceFileName(node.getName());
   ptrSelfPointer = new llvm::GlobalVariable(
-      getModule(), selfType->getPointerTo(STORAGE_ADDR_SPACE), false,
+      getModule(), structTypes["self"]->getPointerTo(STORAGE_ADDR_SPACE), false,
```

Comment:
It would be better to use `getSelfType()` instead of `structTypes["self"]`, in case we want to refactor how the self type is created in the future.
```suggestion
      getModule(), getSelfType()->getPointerTo(STORAGE_ADDR_SPACE), false,
```

Reply:
done